snuba/schemas.py

```diff
@@ -156,48 +143,24 @@
         'offset': {
             'type': 'number',
         },
-        'selected_columns': {
-            'anyOf': [
+        'limitby': {
+            'type': 'array',
+            'items': [
+                {'type': 'number'},
                 {'$ref': '#/definitions/column_name'},
-                {'$ref': '#/definitions/column_list'},
-                {'type': 'array', 'minItems': 0, 'maxItems': 0},
-            ],
-            'default': [],
-        },
-        'sample': {
-            'type': 'number',
-            'min': 0,
-        },
-        # Never add FINAL to queries, enable sampling
-        'turbo': {
-            'type': 'boolean',
-            'default': False,
-        },
-        # Force queries to hit the first shard replica, ensuring the query
-        # sees data that was written before the query. This burdens the
-        # first replica, so should only be used when absolutely necessary.
-        'consistent': {
-            'type': 'boolean',
-            'default': False,
+            ]
         },
-        'debug': {
-            'type': 'boolean',
-        }
     },
-    # Need to select down to the project level for customer isolation and performance
-    'required': ['project'],
     'dependencies': {
         'offset': ['limit'],
         'totals': ['groupby']
     },
     'additionalProperties': False,
-
     'definitions': {
         'column_name': {
             'type': 'string',
             'anyOf': [
-                {'pattern': '^-?[a-zA-Z0-9_.]+$', },
-                {'pattern': r'^-?tags\[[a-zA-Z0-9_.:-]+\]$', },
+                {'pattern': r'^-?[a-zA-Z0-9_.]+(\[[a-zA-Z0-9_.:-]+\])?$', },
```

Comment:
How do you feel about a comment that describes what this regex intends to do? I've found that it's hard to correct a regex without knowing the original intent

Reply:
Sure, that seems worthwhile, will do.

Reply:
I added a description with 2a07e3e1dd05a1f02276ce601a14ae9e289b8757.

Reply:
Also, for later: we should probably consolidate this pattern with these other definitions:

https://github.com/getsentry/snuba/blob/f6ca2b695bfee87ee2313078a874b25f0feb0c6b/snuba/clickhouse.py#L14
https://github.com/getsentry/snuba/blob/f6ca2b695bfee87ee2313078a874b25f0feb0c6b/snuba/datasets/events.py#L26-L27